File: Algorithms/MergeSort/merge_sort.py
def merge_sort(arr):
    if len(arr) > 1:
        mid = len(arr)//2    # getting the mid off array
        L = arr[:mid]
        R = arr[mid:]
        
        merge_sort(L)   # sorting the first sub-array
        merge_sort(R)   # sorting the 2nd sub-array

        i = j = k = 0

        # Copy data to temp arrays L[] and R[]
        while i < len(L) and j < len(R):
            if L[i] < R[j]:
                arr[k] = L[i]
                i += 1
            else:
                arr[k] = R[j]
                j += 1
            k += 1
        
        # checking if any elements was left
        while i < len(L):
            arr[k] = L[i]
            i += 1
            k += 1
        while j < len(R):
            arr[k] = R[j]
            j += 1
            k += 1
        

if __name__ == "__main__":
    arr = [5, 6, 4, 3, 1, 9, 8]
    merge_sort(arr)
    print("Sorted array is: ", arr)

File: Algorithms/MergeSort/merge_sort.js
merge_sort = (arr) => {
    if (arr.length > 1){
        let mid = Math.floor(arr.length / 2)
        let L = arr.slice(0, mid)
        let R = arr.slice(mid, arr.length)

        merge_sort(L)   // sorting the first sub-array
        merge_sort(R)   // sorting the 2nd sub-array
        
        let i = j = k = 0

        // copy data to temp array L[] and R[]
        while (i < L.length && j < R.length){
            if (L[i] < R[j]){
                arr[k] = L[i]
                i++
            }
            else {
                arr[k] = R[j]
                j++
            }
            k++
        }

        // check if any elements was left
        // as above while checking to length, any array might have 1 value more
        while (i < L.length){
            arr[k] = L[i]
            i++
            k++
        }
        while(j < R.length){
            arr[k] = R[j]
            j++
            k++
        }
    }
}

arr = [6, 5, 7, 3, 1, 9, 8]
merge_sort(arr)
console.log(arr)